src/server/cea_608_encoder/byte_pair_generator.py

```diff
@@ -67,7 +67,8 @@ def consume_scenes(scene_list: list) -> list:
     :return: TODO
     """
     scene_data = []
-    
+    validate_scene_ids(scene_list) 
```

Comment:
This call occurs prior to checking whether there currently exists a scene id in a scene. This line: `if 'scene_id' not in scene:` is where we check whether the id even exists but if we call your validation before then we can get a KeyError when you try `scene['scene_id']`

Reply:
Changed it to be called afterwards